birdhouse/components/weaver/docker-compose-extra.yml

```diff
@@ -0,0 +1,129 @@
+version: "3.4"
+
+x-logging:
+  &default-logging
+  driver: "json-file"
+  options:
+    max-size: "200k"
+    max-file: "10"
+
+services:
+
+  proxy:
+    volumes:
+      - ./components/weaver/conf.extra-service.d:/etc/nginx/conf.extra-service.d/weaver:ro
+    links:
+      - weaver
+
+  magpie:
+    volumes:
+      - ./components/weaver/config/magpie/config.yml:/opt/local/src/magpie/config/permissions/weaver-permissions.cfg:ro
+      - ./components/weaver/config/magpie/config.yml:/opt/local/src/magpie/config/providers/weaver-provider.cfg:ro
+
+  # Image 'weaver' is the API side of the application
+  weaver:
+    image: pavics/weaver:${WEAVER_VERSION}-manager
+    container_name: ${WEAVER_MANAGER_NAME}
+    ports:
+      - "4001:4001"
+    environment:
+      HOSTNAME: ${PAVICS_FQDN}
+      FORWARDED_ALLOW_IPS: "*"
+    #env_file:
+    #  - ./config/mongodb/credentials.env
+    # ensure other referenced services are created beforehand to avoid booting timing issues
```

Comment:
We had problems in the past by trusting too much `depends_on`, flyingpigeon in particular can take serious time to boot and be ready. Compose only garantee that the container is started, but not necessarily ready. 
Given the previous comments, the providers are checked at Weaver boot time, so it may ends up with some of these not available without special care.

Reply:
Could use this solution instead, but the boot of the full stack will wait until the longest bird to finish init : 
https://github.com/jwilder/dockerize#waiting-for-other-dependencies